src/transport/Session.h

```diff
@@ -265,15 +265,16 @@ class Session
     // System::SystemClock().GetMonotonicTimestamp() (to indicate "peer is
     // responding to a message it just received") and System::Clock::kZero (to
     // indicate "peer is reaching out to us, not in response to anything").
-    virtual System::Clock::Milliseconds32 GetMessageReceiptTimeout(System::Clock::Timestamp ourLastActivity) const = 0;
+    virtual System::Clock::Milliseconds32 GetMessageReceiptTimeout(System::Clock::Timestamp ourLastActivity,
+                                                                   bool isPeerActive) const = 0;
```

Comment:
This part makes no sense to me.  Whether the _peer_ is active or not has nothing to do with how long it will keep retrying to get a message to _us_.  What matter is whether the peer thinks _we_ are active.

In practice we need to carefully look at the callsites of this function.  I suspect in all of them the peer would be assuming we are active because it's responding to our message, but that needs to be verified carefully.

Reply:
I agree, yes, after checking all calls, i think all of them the peer would be assuming we are active. updated, thanks